src/aws-cpp-sdk-core/include/smithy/identity/signer/built-in/SigV4Signer.h

```diff
@@ -18,24 +18,39 @@ namespace smithy {
      * TODO: refactor into own signer using smithy design
      */
     class AwsSigV4Signer : public AwsSignerBase<AwsCredentialIdentityBase> {
+        
     public:
         using SigV4AuthSchemeParameters = DefaultAuthSchemeResolverParameters;
         explicit AwsSigV4Signer(const SigV4AuthSchemeParameters& parameters)
-            : legacySigner(nullptr, parameters.serviceName.c_str(), *parameters.region)
+            :  m_parameters{parameters}, legacySigner(nullptr, parameters.serviceName.c_str(), *parameters.region)
         {
         }
 
-        SigningFutureOutcome sign(std::shared_ptr<HttpRequest> httpRequest, const AwsCredentialIdentityBase& identity, SigningProperties properties, const AdditionalParameters& additionalParameters) override
+        SigningFutureOutcome sign(std::shared_ptr<HttpRequest> httpRequest, const AwsCredentialIdentityBase& identity, SigningProperties properties) override
         {
-            AWS_UNREFERENCED_PARAM(additionalParameters);
+            const char ALLOCATION_TAG[] = "AwsSigV4Signer";
 
-            Aws::Auth::AWSCredentials legacyCreds(identity.accessKeyId(), identity.secretAccessKey(), *identity.sessionToken(), *identity.expiration());
+            std::shared_ptr<Aws::Auth::AWSCredentials> legacyCreds_sp;
```

Comment:
why are we creating a shared pointer instead of using a stack variable? the only place you use it, you de-reference it to a raw value

Reply:
no why does this need to be a pointer? it looks like a stack variable could just as easily be used, pointer semantics arent required here

Reply:
this will work. Ack

Reply:
cpp 11 doesnt promise copy elision but for a string most modern compilers would do it, i would say prefer stack allocation over heap allocation whenever possible